src/main/kotlin/week1/racingcar/RacingGame.kt

```diff
@@ -1,36 +1,62 @@
 package week1.racingcar
 
-class RacingGame {
+import kotlin.random.Random
 
-    private var numberOfCars = 0
-    private var timesToRace = 0
+class RacingGame {
 
-    fun start() {
-        InputView().startInput().also {
-            this@RacingGame.apply {
-                numberOfCars = it.numberOfCars
-                timesToRace = it.timesToRace
-            }
-            race()
+    fun getInputResult(): InputResult {
+        return InputView().let {
+            it.getValidateRaceInfo(it.startInput())
         }
     }
 
-    private fun race() {
+    fun startGame(inputResult: InputResult) {
+        val cars = inputResult.cars.map { Car(it) }
+        val timesToRace = inputResult.timesToRace
+        race(cars, timesToRace)
+    }
+
+    private fun race(cars: List<Car>, timesToRace: Int) {
         val resultView = ResultView()
-        val cars = List(numberOfCars) { Car() }
-        (0 until timesToRace).forEach { _ ->
-            cars.forEach {
-                if (isPossibleToDrive()) {
-                    it.go()
-                }
-            }
-            resultView.showResult(cars)
+        repeat(timesToRace) {
+            drive(cars)
```

Comment:
메서드를 분리를 조금 과하게 했다고 느껴지는 것 같아요. 🤔
저라면 지금처럼 메서드로 쪼개지않고 아래와 같이 바로 코드를 짰을 것 같아요. 👀

```suggestion
            cars.forEach {
                if (Random.nextInt(COUNT_OF_DIGITS) >= MIN_NUM_TO_GO) {
                    it.go()
                }
            }
```